model/syncQueue/storage/SyncQueueStorageRds.php

```diff
@@ -45,32 +46,99 @@ public function __construct($options = array())
+        $query = $this->getQueryBuilder()
+            ->select([self::PARAM_SYNCHRONIZABLE_ID, self::PARAM_EVENT_TYPE, self::PARAM_SYNC_MIGRATION_ID])
+            ->from(self::TABLE_NAME)
+            ->andWhere(self::PARAM_SYNC_MIGRATION_ID . ' = :migrationId ')
+            ->setParameter('migrationId', $migrationId);
+
+        if (count($types)) {
+            $query->andWhere(self::PARAM_SYNCHRONIZABLE_TYPE . ' IN (:sync_type)')
+                ->setParameter('sync_type', $types, Connection::PARAM_STR_ARRAY);
+        }
+
+        $query
+            ->groupBy([self::PARAM_SYNCHRONIZABLE_ID, self::PARAM_EVENT_TYPE, self::PARAM_SYNC_MIGRATION_ID])
+            ->setMaxResults($limit);
+
+        return $query->execute()->fetchAll();
     }
 
-    public function getAll($limit = 1000, $offset = 0)
+    /**
+     * @param array $types
+     * @param int $limit
+     * @return array|mixed[]
+     */
+    public function getAggregatedQueued(array $types = [], $limit = 5000)
     {
-        $sql = 'SELECT * FROM ' . self::TABLE_NAME . ' ORDER BY ' . self::PARAM_ID . ' LIMIT ? OFFSET ?';
-        $parameters = [0];
-        $stmt = $this->getPersistence()->query($sql, $parameters);
-        return $stmt->fetchAll(PDO::FETCH_ASSOC);
+        return $this->getMigrationData(0, $types, $limit);
+    }
+
+    public function getAll($limit = 10000, $offset = 0)
+    {
+        $query = $this->getQueryBuilder()
+            ->select('*')
+            ->from(self::TABLE_NAME)
+            ->orderBy(self::PARAM_CREATED_AT)
+            ->setMaxResults($limit);
+
+        return $query->execute()->fetchAll();
+    }
+
+    /**
+     * Returns the QueryBuilder
+     *
+     * @return QueryBuilder
+     */
+    private function getQueryBuilder()
+    {
+        return $this->getPersistence()->getPlatform()->getQueryBuilder();
     }
 
     public function insert(array $action)
     {
         return $this->getPersistence()->insert(self::TABLE_NAME, $action);
     }
 
-    public function setSyncId($id)
+    public function setMigrationId($migrationId, $queuedTasks = [])
+    {
+        foreach ($queuedTasks as $queuedTask) {
+            $qb = $this->getPersistence()->getPlatForm()->getQueryBuilder();
+            $qb->update(static::TABLE_NAME)
+                ->set(static::PARAM_SYNC_MIGRATION_ID, ':migrationId')
+                ->set(static::PARAM_UPDATED_AT, ':date')
+                ->andWhere(static::PARAM_SYNCHRONIZABLE_ID . ' =:syncId')
+                ->andWhere(static::PARAM_EVENT_TYPE . ' =:eventType')
+                ->setParameter('migrationId', $migrationId)
+                ->setParameter('date', date('Y-m-d H:i:s'))
+                ->setParameter('syncId', $queuedTask[self::PARAM_SYNCHRONIZABLE_ID])
+                ->setParameter('eventType', $queuedTask[self::PARAM_EVENT_TYPE])
+                ->execute();
+        }
+        return true;
+    }
+
+    public function isSynchronized($eventType = '', $synchronizedIds = [])
```

Comment:
I think it's better to rename `synchronizedIds` to `synchronizableIds`, how it's called in database, to avoid confusions

Reply:
it's a different thing. Synchronizable - means a resource which could be synchronized. Synchronized - is a state of this resource.